zcash_primitives/src/transaction/components/sapling.rs

```diff
@@ -109,13 +110,13 @@ fn read_cmu<R: Read>(mut reader: R) -> io::Result<ExtractedNoteCommitment> {
 
 /// Consensus rules (§7.3) & (§7.4):
 /// - Canonical encoding is enforced here
-pub fn read_base<R: Read>(mut reader: R, field: &str) -> io::Result<jubjub::Base> {
+pub fn read_base<R: Read>(mut reader: R, _field: &str) -> io::Result<jubjub::Base> {
     let mut f = [0u8; 32];
     reader.read_exact(&mut f)?;
     Option::from(jubjub::Base::from_repr(f)).ok_or_else(|| {
         io::Error::new(
             io::ErrorKind::InvalidInput,
-            format!("{} not in field", field),
+            "base value not a valid field element",
```

Comment:
Non-blocking: https://stackoverflow.com/questions/50200268/how-can-i-use-the-format-macro-in-a-no-std-environment has some alternatives to `format!` that can be used in a no_std environment.

Reply:
This change is because `core2::io::Error::new` takes a `&'static str` instead of a `String` here, and there's no way to format to a `&'static str`.